tests/phpunit/tests/media.php

```diff
@@ -5553,6 +5553,42 @@ public function test_wp_get_loading_optimization_attributes_if_loading_attr_pres
 		);
 	}
 
+	/**
+	 * @ticket 61436
+	 *
+	 * @covers ::wp_get_loading_optimization_attributes
+	 */
+	public function test_wp_get_loading_optimization_attributes_image_src_matched() {
+		add_filter(
+			'wp_get_loading_optimization_attributes',
+			static function ( $loading_attrs, $tag_name, $attr ) {
+				if (
+					'img' === $tag_name &&
+					isset( $attr['src'] ) &&
+					'https://example.org/a-specific-image.jpg' === $attr['src']
+				) {
+					$loading_attrs['fetchpriority'] = 'high';
+				}
+				return $loading_attrs;
+			},
+			10,
+			3
+		);
+		$attr = array(
+			'src' => 'https://example.org/a-specific-image.jpg',
+		);
+
+		// fetchpriority is high when image src is matched.
+		$this->assertSameSetsWithIndex(
+			array(
+				'decoding'      => 'async',
+				'fetchpriority' => 'high',
+			),
+			wp_get_loading_optimization_attributes( 'img', $attr, 'the_content' ),
```

Comment:
We need to call `wp_img_tag_add_loading_optimization_attrs()` instead, since that's where the bug fix is being made.